### PR TITLE
Updated documentation of arithmetic functions

### PR DESCRIPTION
**Pull request by matthewryanwells**
### Description
There were a few mistakes in the arithmetic function documentation that I have fixed.
- Added missing arithmetic operators
- Added missing data types
- Added missing function
- Fixed incorrect function usage

### Issues Resolved
This PR is not associated with any issues

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _search-plugins/sql/functions.md [hunk 1]
```diff
@@ -33,20 +33,24 @@ ln | `ln(number T) -> double` | `SELECT ln(10) FROM my-index LIMIT 1`
 log | `log(number T) -> double` or `log(number T, number) -> double` | `SELECT log(10) FROM my-index LIMIT 1`
```
Comment (GumpacG) [inline _search-plugins/sql/functions.md:33]: This one too

File: _search-plugins/sql/functions.md [hunk 2]
```diff
@@ -33,20 +33,24 @@ ln | `ln(number T) -> double` | `SELECT ln(10) FROM my-index LIMIT 1`
...
 log2 | `log2(number T) -> double` | `SELECT log2(10) FROM my-index LIMIT 1`
 log10 | `log10(number T) -> double` | `SELECT log10(10) FROM my-index LIMIT 1`
-mod | `mod(number T, number) -> T` | `SELECT modulus(2, 3) FROM my-index LIMIT 1`
-multiply | `multiply(number T, number) -> number` | `SELECT multiply(2, 3) FROM my-index LIMIT 1`
+mod | `mod(number T, number) -> T` | `SELECT mod(2, 3) FROM my-index LIMIT 1`
```
Comment (GumpacG) [inline _search-plugins/sql/functions.md]: `mod(number T, number T)`

File: _search-plugins/sql/functions.md (hunk 1 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md:33]: I believe that this one doesn't need to be updated as calling a `log(number T)` defaults the log to be base e

File: _search-plugins/sql/functions.md [hunk 3]
```diff
@@ -33,20 +33,24 @@ ln | `ln(number T) -> double` | `SELECT ln(10) FROM my-index LIMIT 1`
...
+modulus | `modulus(number T, number T) -> T` | `SELECT modulus(2, 3) FROM my-index LIMIT 1`
+multiply | `multiply(number T, number T) -> T` | `SELECT multiply(2, 3) FROM my-index LIMIT 1`
 pi | `pi() -> double` | `SELECT pi() FROM my-index LIMIT 1`
-pow | `pow(number T) -> T` or `pow(number T, number) -> T` | `SELECT pow(2, 3) FROM my-index LIMIT 1`
-power | `power(number T) -> T` or `power(number T, number) -> T` | `SELECT power(2, 3) FROM my-index LIMIT 1`
+pow | `pow(number T, number) -> T` | `SELECT pow(2, 3) FROM my-index LIMIT 1`
```
Comment (GumpacG) [inline _search-plugins/sql/functions.md]: `pow(number T, number T)`

File: _search-plugins/sql/functions.md (hunk 1 above)
Comment (GumpacG) [reply _search-plugins/sql/functions.md:33]: Does `log(number T, number)` need to be `log(number T, number T)`?

Comment (matthewryanwells) [reply _search-plugins/sql/functions.md:33]: The `log` method can either have one or two parameters, if it has one it defaults to the natural log ln, and if you put in two parameters then one of the parameters is set to be the base

File: _search-plugins/sql/functions.md (hunk 3 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: Good catch, fixed in https://github.com/Bit-Quill/documentation-website/pull/3/commits/b0ab2b66a4899b61f9f82b4b551bc0aabe46a31c

File: _search-plugins/sql/functions.md (hunk 2 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: Fixed in https://github.com/Bit-Quill/documentation-website/pull/3/commits/b0ab2b66a4899b61f9f82b4b551bc0aabe46a31c

**Review by GumpacG — APPROVED** [review]

File: _search-plugins/sql/functions.md [hunk 4]
```diff
@@ -33,20 +33,24 @@ ln | `ln(number T) -> double` | `SELECT ln(10) FROM my-index LIMIT 1`
...
 rint | `rint(number T) -> T` | `SELECT rint(1.5) FROM my-index LIMIT 1`
 round | `round(number T) -> T` | `SELECT round(1.5) FROM my-index LIMIT 1`
 sign | `sign(number T) -> T` | `SELECT sign(1.5) FROM my-index LIMIT 1`
 signum | `signum(number T) -> T` | `SELECT signum(0.5) FROM my-index LIMIT 1`
 sqrt | `sqrt(number T) -> T` | `SELECT sqrt(0.5) FROM my-index LIMIT 1`
 strcmp | `strcmp(string T, string T) -> T` | `SELECT strcmp('hello', 'hello') FROM my-index LIMIT 1`
-subtract | `subtract(number T, number) -> T` | `SELECT subtract(3, 2) FROM my-index LIMIT 1`
+subtract | `subtract(number T, number T) -> T` | `SELECT subtract(3, 2) FROM my-index LIMIT 1`
 truncate | `truncate(number T, number T) -> T` | `SELECT truncate(56.78, 1) FROM my-index LIMIT 1`
++ | `number [op] number -> number` | `SELECT 1 + 100 FROM my-index LIMIT 1`
```
Comment (Yury-Fridlyand) [inline _search-plugins/sql/functions.md]: Shouldn't we replace `[op]` with exact operation since they are listed one per line?

**Review by Yury-Fridlyand — APPROVED** [review]

File: _search-plugins/sql/functions.md (hunk 4 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: That's a good point, I was just copying the syntax that was already there but putting the operation in makes more sense.

Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: Updated in https://github.com/Bit-Quill/documentation-website/pull/3/commits/55085495c1c77f75c253cabbb1388d9d3d1d1e13

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**